res/controllers/Hercules-DJ-Control-Instinct-P8-scripts.js

```diff
@@ -0,0 +1,261 @@
+    "inputs": {
+        0x30: { "name": "jog", "channel": 1, "group": "[Channel1]"},
+        0x32: { "name": "jog", "channel": 2, "group": "[Channel2]"},
+        0x31: { "name": "pitch", "channel": 1, "group": "[Channel1]"},
+        0x33: { "name": "pitch", "channel": 2, "group": "[Channel2]"},
+    }
+};
+
+
+DJControlInstinctP8.scratch = function(midino, control, value, status, group) {
+
+    // Normal scratch function
+    if (value) {
+      if (scratchMode === 0) {
+          // Enable the scratch mode on the corrisponding deck and start the timer
+          scratchMode = 1;
+          scratchTimer = engine.beginTimer(scratchResetTime, DJControlInstinctP8.wheelOnOff);
+          midi.sendShortMsg(0x90, 45, 0x7F); // Switch-on the Scratch led
+          engine.setValue("[Channel1]", "keylock", 0);
+          engine.setValue("[Channel2]", "keylock", 0);
+      } else {
+          // Disable the scratch mode on the corrisponding deck and stop the timer
+          scratchMode = 0;
+          engine.stopTimer(scratchTimer);
+          midi.sendShortMsg(0x90, 45, 0x00); // Switch-off the Scratch led (45=2D)
+      }
+    }
+};
+
+// This function is called every "scratchResetTime" seconds and checks if the wheel was moved in the previous interval
+// (every interval last "scratchResetTime" seconds). If the wheel was moved enables the scratch mode, else disables it.
+// In this way I have made a simple workaround to simulate the touch-sensitivity of the other controllers.
+DJControlInstinctP8.wheelOnOff = function() {
+    // Wheel Deck A / Channel 1
+    if (wheelMove[0]) {
+        engine.scratchEnable(1, 128, standardRpm, alpha, beta);
+    }
+    else { engine.scratchDisable(1);
+    }
+	    wheelMove[0] = 0;
+    
+
+    //Wheel Deck B / Channel 2
+    if (wheelMove[1]) {
+        engine.scratchEnable(2, 128, standardRpm, alpha, beta);
+    }
+    else {engine.scratchDisable(2);
+    }
+	    wheelMove[1] = 0;
+    
+};
+
+DJControlInstinctP8.jogWheel = function (midino, control, value, status, group) {
+
+    var deck = (group == "[Channel1]") ? deckA : deckB;
+
+    // This function is called everytime the jog is moved
+
+	var direction = 1;
+	if (value !== 0x01) {
+    	direction = -1;
+	}
+
+	if (scratchMode) {
+    	engine.scratchTick(deck, direction);
+    	wheelMove[deck - 1] = 1;
+	} else {
+    	engine.setValue("[Channel" + deck + "]", "jog", jogSensitivity * direction);
+	}
+
+};
+
+DJControlInstinctP8.jogWheelHelper = function(n) {
+    engine.scratchDisable(n);
+    scratch_timer_on[n] = false;
+};
+
+// Pitch is adjusted by holding down shift and turning the jog wheel.
+DJControlInstinctP8.pitch = function (group, control, value, status) {
+    var input = DJControlInstinctP8.controls.inputs[control];
+
+    // If the high bit is 1, convert to a negative number
+    if (value & 0x40) {
+        value = value - 0x80;
+    }
+    var delta = Math.abs(Math.pow(Math.abs(value), 2) / 1000.0);
+//    if (value < 0) {

```

Comment:
I think I gave you bad advice on this block of code previously... sorry about that, you should probably revert to original.

```javascript
    var delta = Math.pow(Math.abs(value), 2) / 1000.0;
    if (value < 0) {
        delta = -delta;
    }
```

Sorry again! I misread.